src/main.rs

```diff
@@ -130,37 +159,23 @@ fn print_error(writer: &mut dyn io::Write, err: &Error) -> Result<(), Error> {
     Ok(())
 }
 
-fn run_lexer_test<L, O>(lexer: L, out: &mut O) -> Result<(), Error>
```

Comment:
This function was split up into `run_compiler` (looping) and `write_token` (filtering whitespace and comments, and `writeln`).

In consequence, `lexer_test_with_tokens`had to be fixed by duplicating the iteration code there (`tokens.into_iter()`).
This makes it a white-box test, whereas it was black-box test before.

I assume this was done because `--lexertest` now has two outputs (`stdout` and the `Context.write`).
The new `run_lexer_test` should have 2 writers as parameters.

Reply:
IMO the `run_lexer_test` is weird. @flip1995 had the same opinion. It only abstracts away a simple for loop.